app/user.go

```diff
@@ -992,19 +992,40 @@ func (a *App) UpdateActive(user *model.User, active bool) (*model.User, *model.A
 	ruser := userUpdate.New
 
 	if !active {
-		if err := a.userDeactivated(ruser); err != nil {
+		if err := a.userDeactivated(ruser.Id); err != nil {
 			return nil, err
 		}
 	}
 
-	a.invalidateUserChannelMembersCaches(user)
+	a.invalidateUserChannelMembersCaches(user.Id)
 	a.InvalidateCacheForUser(user.Id)
 
 	a.sendUpdatedUserEvent(*ruser)
 
 	return ruser, nil
 }
 
+func (a *App) DeactivateGuests() *model.AppError {
```

Comment:
If you make this a function of the server then you can avoid the use of the fake app above.

Reply:
thanks, I'll do it

Reply:
moving that function to the server is going to be a pain because I need to do things related to websockets notifications, and all that is at App layer.

For now I don't see any reason to not use the FakeApp, at least in the initialization.

I going to create a new app instance in using the same approach used in the handlers based on the runtime server instance, it should be enough (I guess).